src/rain-report/rain-report.controller.ts

```diff
@@ -2,25 +2,33 @@ import {
   Body,
   Controller,
   Get,
+  Headers,
   HttpCode,
   HttpStatus,
   Post,
+  UseGuards,
 } from '@nestjs/common';
-import { RainReportService } from './rain-report.service';
+
+import { AuthGuard, AUTH_HEADER } from '../auth/auth.guard';
 import { CreateRainReportDto } from './dto/create-rain-report.dto/create-rain-report.dto';
+import { RainReportService } from './rain-report.service';
 
 @Controller('data')
+@UseGuards(AuthGuard)
```

Comment:
`AuthGuard` implementation. Applied at controller level to guard both endpoints.